pkg/cmd/pulumi/pulumi.go

```diff
@@ -186,55 +267,94 @@ func NewPulumiCmd() *cobra.Command {
 
-	// Common commands:
-	//     - Getting Started Commands:
-	cmd.AddCommand(newNewCmd())
-	//     - Deploy Commands:
-	cmd.AddCommand(newUpCmd())
-	cmd.AddCommand(newPreviewCmd())
-	cmd.AddCommand(newDestroyCmd())
-	cmd.AddCommand(newWatchCmd())
-	//     - Stack Management Commands:
-	cmd.AddCommand(newStackCmd())
-	cmd.AddCommand(newConfigCmd())
-	//     - Service Commands:
-	cmd.AddCommand(newLoginCmd())
-	cmd.AddCommand(newLogoutCmd())
-	cmd.AddCommand(newWhoAmICmd())
-	//     - Policy Management Commands:
-	cmd.AddCommand(newPolicyCmd())
-	//     - Advanced Commands:
-	cmd.AddCommand(newCancelCmd())
-	cmd.AddCommand(newImportCmd())
-	cmd.AddCommand(newRefreshCmd())
-	cmd.AddCommand(newStateCmd())
-	//     - Other Commands:
-	cmd.AddCommand(newLogsCmd())
-	cmd.AddCommand(newPluginCmd())
-	cmd.AddCommand(newVersionCmd())
-	cmd.AddCommand(newConsoleCmd())
-	cmd.AddCommand(newAboutCmd())
-	cmd.AddCommand(newSchemaCmd())
-	cmd.AddCommand(newOrgCmd())
-
-	// Less common, and thus hidden, commands:
-	cmd.AddCommand(newGenCompletionCmd(cmd))
-	cmd.AddCommand(newGenMarkdownCmd(cmd))
-
-	// We have a set of commands that are still experimental and that are hidden unless PULUMI_EXPERIMENTAL is set
-	// to true.
-	cmd.AddCommand(newQueryCmd())
-	cmd.AddCommand(newConvertCmd())
-
-	// We have a set of options that are useful for developers of pulumi that are hidden unless PULUMI_DEBUG_COMMANDS is
-	// set to true.
-	cmd.PersistentFlags().StringVar(&tracingHeaderFlag, "tracing-header", "",
-		"Include the tracing header with the given contents.")
+	setCommandGroups(cmd, []commandGroup{
+
+		// Common commands:
+		{
+			Name: "Deployment Commands",
+			Commands: []*cobra.Command{
+				newNewCmd(),
+				newConfigCmd(),
+				newUpCmd(),
+				newDestroyCmd(),
+				newPreviewCmd(),
+				newCancelCmd(),
+			},
+		},
+		{
+			Name: "Stack Management Commands",
+			Commands: []*cobra.Command{
+				newStackCmd(),
+				newConsoleCmd(),
+				newImportCmd(),
+				newRefreshCmd(),
+				newStateCmd(),
+			},
+		},
+		{
+			Name: "Service Commands",
+			Commands: []*cobra.Command{
+				newLoginCmd(),
+				newLogoutCmd(),
+				newWhoAmICmd(),
+				newOrgCmd(),
+			},
+		},
+		{
+			Name: "Policy Management Commands",
+			Commands: []*cobra.Command{
+				newPolicyCmd(),
+			},
+		},
+		{
+			Name: "Plugin Commands",
+			Commands: []*cobra.Command{
+				newPluginCmd(),
+				newSchemaCmd(),
+			},
+		},
+		{
+			Name: "Other Commands",
+			Commands: []*cobra.Command{
+				newVersionCmd(),
+				newAboutCmd(),
+			},
+		},
 
-	cmd.AddCommand(newViewTraceCmd())
-	cmd.AddCommand(newConvertTraceCmd())
+		// Less common, and thus hidden, commands:
+		{
+			Name: "Hidden Commands",
+			Commands: []*cobra.Command{
+				newGenCompletionCmd(cmd),
+				newGenMarkdownCmd(cmd),
+			},
+		},
 
-	cmd.AddCommand(newReplayEventsCmd())
+		// We have a set of commands that are still experimental
+		//     hidden unless PULUMI_EXPERIMENTAL is set to true.
```

Comment:
Is this remark supposed to be true? I see all except `query` even though I don't have this flag set.

Reply:
That comment is outdated. I'll add a comment next to the applicable commands.